src/main/java/LinkedBinarySearchTree.java

```diff
@@ -432,50 +406,47 @@ public String toString() {
     }
 
     /**
-     * A Node class for a binary linked tree structure. Each node
-     * contains a reference to data of type T, which may be
-     * null, and a reference to the left and right child
-     * binary linked nodes, which may also be null.
-     * <p>
-     * This class is a static nested class since the node
-     * class is only needed for the implementation of the
-     * LinkedBinarySearchTree.
+     * A node class for a linked binary tree structure. Each node contains a nullable reference to data of type T, and a
+     * reference to the left and right child nodes, which may be null references.
      *
-     * @param <T> Type of the data being stored in the node
+     * @param <T> Type of the data being stored in the node.
      */
     private static class Node<T> {
-
         private T data;
         private Node<T> left;
         private Node<T> right;
 
-        private Node(T data) {
+        public Node() {
+            this(null);
+        }
+
+        public Node(T data) {
             this.data = data;
             this.left = null;
             this.right = null;
         }
 
-        private T getData() {
+        public T getData() {
```

Comment:
Why are these public now? The class itself is private, it should never have anything public unless strictly required.

Reply:
Good catch. They shouldn't be. I did a lazy copy paste.